specification/network/resource-manager/Microsoft.Network/preview/2021-02-01-preview/networkSecurityPerimeter.json

```diff
@@ -351,6 +852,261 @@
+        }
+      },
+      "allOf": [
+        {
+          "$ref": "./network.json#/definitions/Resource"
+        }
+      ],
+      "description": "The NSP access rule resource"
+    },
+    "NspAccessRuleListResult": {
+      "type": "object",
+      "properties": {
+        "value": {
+          "type": "array",
+          "items": {
+            "$ref": "#/definitions/NspAccessRule"
+          },
+          "description": "Gets a page of NSP access rule"
+        },
+        "nextLink": {
+          "type": "string",
+          "description": "Gets the URL to get the next page of results."
+        }
+      },
+      "description": "Result of the request to list NSP access rules. Contains a list of NSP access rules and a URL link to get the next set of results."
+    },
+    "NspAccessRuleProperties": {
+      "type": "object",
+      "properties": {
+        "provisioningState": {
+          "readOnly": true,
+          "$ref": "./network.json#/definitions/ProvisioningState",
+          "description": "The provisioning state of the scope assignment resource."
+        },
+        "direction": {
+          "type": "string",
+          "enum": [
+            "Inbound",
+            "Outbound"
+          ],
+          "x-ms-enum": {
+            "name": "AccessRuleDirection",
+            "modelAsString": true
+          },
+          "description": "Direction that specifies whether the access rules is inbound/outbound."
+        },
+        "addressPrefixes": {
+          "type": "array",
+          "items": {
+            "type": "string"
+          },
+          "description": "Inbound address prefixes (IPv4/IPv6)"
+        },
+        "fullyQualifiedDomainNames": {
+          "type": "array",
+          "items": {
+            "type": "string"
+          },
+          "description": "Outbound rules fully qualified domain name format."
+        },
+        "subscriptions": {
+          "type": "array",
+          "items": {
+            "type": "string"
+          },
+          "description": "Subscription id in the ARM id format."
+        },
+        "networkSecurityPerimeters": {
+          "type": "array",
+          "items": {
+            "x-ms-client-flatten": true,
+            "$ref": "#/definitions/PerimeterBasedAccessRule"
+          },
+          "description": "Inbound rule specified by the perimeter id."
+        }
+      },
+      "description": "Properties of NSP access rule."
+    },
+    "PerimeterBasedAccessRule": {
+      "type": "object",
+      "properties": {
+        "id": {
+          "type": "string",
+          "description": "NSP id in the ARM id format."
+        },
+        "perimeterGuid": {
+          "readOnly": true,
+          "type": "string",
+          "description": "Resource guid of the NSP supplied."
+        },
+        "location": {
+          "readOnly": true,
+          "type": "string",
+          "description": "Location of the NSP supplied."
+        }
```

Comment:
curious why do you need location here ?

Reply:
We need it in the response. This field is used by the partner RPs for querying further.

Reply:
Why do they need location to query a resource if the full id is present already ?